### PR TITLE
Polygon map start boxes

### PR DESCRIPTION
**Pull request by Beherith**
Feed in the points of a polygon in clockwise order for joy.

image

### CODE REVIEW

**Comment by Beherith** [comment]
its draft cause it still has some debug code in.

**Comment by WatchTheFort** [comment]
@Beherith What is the status of this?

**Comment by Beherith** [comment]
Ready to go! Unless @Damgam  wants to do some norushtimer stuff

File: luaui/Widgets/map_startbox.lua [hunk 1]
```diff
@@ -213,6 +213,214 @@ local function drawFormList()
...
+			pingData = {0,0,0,-10000}, -- x,y,z, time
+		},
+		shaderName = "Start Polygons GL4",
+		shaderConfig = {
+			ALPHA = 0.5,
+			NUM_POLYGONS = 0,
+			NUM_POINTS = 0,
+			MINY = minY - 10,
+			MAXY = maxY + 100,
+			MAX_STEEPNESS = 0.5877, -- 45 degrees yay (cos is 0.7071? (54 degrees, so cos of that is 0.5877	)
```
Comment (sprunk) [inline luaui/Widgets/map_startbox.lua:294]:
```suggestion
			MAX_STEEPNESS = math.cos(math.rad(54)),
```

File: luaui/Widgets/map_startpolygon_gl4.lua [hunk 2]
```diff
@@ -0,0 +1,227 @@
+function widget:GetInfo()
```
Comment (sprunk) [inline luaui/Widgets/map_startpolygon_gl4.lua:1]: This whole file seems to be a copy-paste of the change to `map_startbox.lua` above.

File: luaui/Widgets/Shaders/map_startpolygon_gl4.frag.glsl [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+	}
+
+
+	// Note that now we have the distance to the closest box in closestbox
+	// and the distance to the most distant box in furthestbox
+
+	// First we color based on their distance
+	fragColor.rgba = vec4(mycolor * sin(closestbox*3 / (40/3.14)), 0.5);
+	//fragColor.rgba = vec4(mycolor*0.5, 0.5);
+	if (timeInfo.x < 150) {
```
Comment (sprunk) [inline luaui/Widgets/Shaders/map_startpolygon_gl4.frag.glsl:337]: Consider using seconds (including below as appropriate)
```suggestion
	if (timeInfo.z < 5) {
```

File: luaui/Widgets/map_startbox.lua [hunk 4]
```diff
@@ -213,6 +213,214 @@ local function drawFormList()
...
+local scavengerStartBoxTexture = "LuaUI/Images/scav-tileable_v002_small.tga"
+
+local raptorStartBoxTexture = "LuaUI/Images/rapt-tileable_v002_small.tga"
+
+local getMiniMapFlipped = VFS.Include("luaui/Widgets/Include/minimap_utils.lua").getMiniMapFlipped
+
+local scavengerAITeamID = 999
+local raptorsAITeamID = 999
+local scavengerAIAllyTeamID
+local raptorsAIAllyTeamID
```
Comment (sprunk) [inline luaui/Widgets/map_startbox.lua:241]: Do the teamIDs need the magic 999 initialisation? There doesn't seem to be any arithmetic done on them
```suggestion
local scavengerAITeamID
local raptorsAITeamID
local scavengerAIAllyTeamID
local raptorsAIAllyTeamID
```

File: luaui/Widgets/map_startpolygon_gl4.lua (hunk 2 above)
Comment (Beherith) [reply luaui/Widgets/map_startpolygon_gl4.lua:1]: Its the other way around.

File: luaui/Widgets/Shaders/map_startpolygon_gl4.frag.glsl (hunk 3 above)
Comment (Beherith) [reply luaui/Widgets/Shaders/map_startpolygon_gl4.frag.glsl]: It was leftover trash anyway

**Pull request merged**